test/rhsmlib_test/test_collector.py

```diff
@@ -43,3 +44,16 @@ def test_get_arch(self):
     def test_get_platform_specific_info_provider(self):
         info_provider = firmware_info.get_firmware_collector(arch=platform.machine())
         self.assertTrue(info_provider is not None)
+
+
+class GetNonDmiUuid(unittest.TestCase):
+    def test_get_aarch64_firmware_collector(self):
+        firmware_provider_class = firmware_info.get_firmware_collector(arch='aarch64')
+        self.assertTrue(isinstance(firmware_provider_class, UuidFirmwareInfoCollector))
```

Comment:
I would add one more unit test for the case the `/sys/devices/virtual/dmi/id/product_uuid` is not available or it is not possible to read it due to access permissions.